File: 6.py
# -*- coding:utf-8 -*-
# 题目: 旋转数组的最小数字


class Solution:
    def minNumberInRotateArray(self, rotateArray):
        # write code here
        i = 0
        while rotateArray[i] >= rotateArray[0] and i < len(rotateArray):
            i += 1
        i = i % len(rotateArray)
        return rotateArray[i]


obj = Solution()
print(obj.minNumberInRotateArray([3, 4, 5, 1, 2]))

File: 0-9/6.go
package main

import "fmt"

/**
 *
 * @param rotateArray int整型一维数组
 * @return int整型
 */
func minNumberInRotateArray(rotateArray []int) int {
	// write code here
	result := 0
	if len(rotateArray) == 0 {
		return result
	}
	if rotateArray[0] < rotateArray[len(rotateArray)-1] {
		result = rotateArray[0]
	} else {
		i := 0
		for i < len(rotateArray)-1 && rotateArray[i] <= rotateArray[i+1] {
			i++
		}
		result = rotateArray[i+1]
	}

	return result
}

func main() {
	fmt.Println(minNumberInRotateArray([]int{3, 4, 5, 1, 2}))
}
